src/components/search/utils/useGetSearchResults.ts

```diff
@@ -1,12 +1,75 @@
-import { Result, useSearchState } from "@yext/search-headless-react";
+import { useState, useEffect } from "react";
+import {
+  Result,
+  SearchCore,
+  useSearchState,
+  useSearchActions,
+} from "@yext/search-headless-react";
+
+async function fetchAll<T>(core: SearchCore, verticalKey: string) {
+  async function fetchPage(offset: number) {
+    const res = await core.verticalSearch({
+      query: "",
+      verticalKey: verticalKey,
+      limit: 50,
+      offset,
+      retrieveFacets: false,
+      skipSpellCheck: true,
+    });
+    return res;
+  }
+
+  let offset = 0;
+  let done = false;
+  const out = [];
+
+  while (!done) {
+    const res = await fetchPage(offset);
+    out.push(...(res.verticalResults.results as Result<T>[]));
+    if (out.length === res.verticalResults.resultsCount) {
+      done = true;
+    }
+    offset += 50;
+  }
+
+  return out;
+}
 
 // Get the results from the search state and cast to the given type.
-export function useGetSearchResults<T>(displayAllOnNoResults?: boolean) {
+export function useGetSearchResults<T>(
+  displayAllOnNoResults?: boolean,
+  allResultsOnLoad?: boolean,
+  allResultsLoadedCallback?: () => void
+) {
+  const [allResults, setAllResults] = useState<Result<T>[]>([]);
+  const [allResultsLoaded, setAllResultsLoaded] = useState(false);
+  const actions = useSearchActions();
+  const state = useSearchState((s) => s);
+  const initialLoad = !state.query.queryId;
+
+  useEffect(() => {
+    if (!allResultsOnLoad || allResultsLoaded || !state.vertical.verticalKey)
```

Comment:
Should this also check that a fetch is not already in progress?

Reply:
I think I'm actually checking more than we need here, the effect hook only runs once because of the empty dependencies array, updated to simplify.